File: LeetCodeProblemsGo/LeetCode1975MaximumMatrixSum/LeetCode1975MaximumMatrixSum.go
package LeetCode1975MaximumMatrixSum

import "math"

func maxMatrixSum(matrix [][]int) int64 {
	var maxSum int64
	minAbsolute := math.MaxInt32
	negativeCount := 0
	for _, row := range matrix {
		for _, num := range row {
			if num < 0 {
				negativeCount++
				num *= -1
			}
			maxSum += int64(num)
			minAbsolute = min(minAbsolute, num)
		}
	}
	if negativeCount%2 == 1 {
		maxSum -= 2 * int64(minAbsolute)
	}
	return maxSum
}

func Calculate() {
	matrix := [][]int{{1, 2, 3}, {-1, -2, -3}, {1, 2, 3}}
	println(maxMatrixSum(matrix))
}

File: LeetCodeProblemsCMake/Problems/LeetCode1975MaximumMatrixSum.cpp
//
// Created by saeed on 1/5/26.
//

#include "LeetCode1975MaximumMatrixSum.h"

#include <stdint.h>
using namespace std;

long long LeetCode1975MaximumMatrixSum::maxMatrixSum(vector<vector<int> > &matrix) {
    long long sum{};
    int negativeCount{};
    int minAbsolute= INT32_MAX;
    for (int i=0,e=matrix.size(); i<e; i++) {
        for (int j=0,e=matrix[0].size(); j<e; j++) {
            if (matrix[i][j] <0) {
                ++negativeCount;
                matrix[i][j]*=-1;
            }
            sum+=matrix[i][j];
            minAbsolute=min(minAbsolute, matrix[i][j]);
        }
    }
    if (negativeCount%2==1) {
        sum-= 2*minAbsolute;
    }
    return sum;
}

